tests/workspace/test_workflows.py

```diff
@@ -65,49 +73,37 @@ def test_invalid_target(args):
     [("main", 5, WORKFLOWS_MAIN), (None, 6, WORKFLOWS)],
 )
 def test_get_workflows(branch, num_workflows, workflows):
+    # get_workflows is called in __init__, so create the instance here
     httpretty.register_uri(
         httpretty.GET,
         uri="https://api.github.com/repos/opensafely-core/airlock/actions/workflows?format=json",
         match_querystring=True,
         body=Path("tests/workspace/workflows.json").read_text(),
     )
-    reporter = jobs.WorkflowReporter("opensafely-core", "airlock", branch=branch)
+    reporter = jobs.RepoWorkflowReporter("opensafely-core", "airlock", branch=branch)
     assert len(reporter.workflows) == num_workflows
     assert reporter.workflows == workflows
 
 
-@httpretty.activate(allow_net_connect=False)
 @pytest.mark.parametrize(
-    "branch, uri_param",
-    [("main", "branch=main&"), (None, "")],
+    "branch, params_str",
+    [("main", "branch=main&format=json"), (None, "format=json")],
 )
-def test_get_all_runs(mock_airlock_reporter, branch, uri_param):
+def test_get_all_runs(mock_airlock_reporter, branch, params_str):
     mock_airlock_reporter.branch = branch  # Overwrite branch to test branch=None
-    httpretty.register_uri(
-        httpretty.GET,
-        f"https://api.github.com/repos/opensafely-core/airlock/actions/runs?{uri_param}format=json",
-        body=Path("tests/workspace/runs.json").read_text(),
-        match_querystring=True,
-    )
     all_runs = mock_airlock_reporter.get_all_runs()
+    assert httpretty.latest_requests()[-1].path.split("?")[1] == params_str
```

Comment:
There is `httpretty.last_request()` that will give you `httpretty.latest_requests()[-1]`
There's also a querystring attribute which probably gets you the params (haven't actually checked though)
```
httpretty.last_request().querystring
```